integration-tests/src/test/java/oracle/weblogic/kubernetes/ItAuxDomainImplicitUpgrde.java

```diff
@@ -0,0 +1,261 @@
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+
+    // get a new unique opNamespace
+    logger.info("Creating unique namespace for Operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Creating unique namespace for domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // Create the repo secret to pull the image
+    // this secret is used only for non-kind cluster
+    createOcirRepoSecret(domainNamespace);
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    adminSecretName = "weblogic-credentials";
+    createSecretWithUsernamePassword(adminSecretName, domainNamespace, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    encryptionSecretName = "encryptionsecret";
+    createSecretWithUsernamePassword(encryptionSecretName, domainNamespace,
+        ENCRYPION_USERNAME_DEFAULT, ENCRYPION_PASSWORD_DEFAULT);
+
+    // build app
+    assertTrue(buildAppArchive(defaultAppParams()
+            .srcDirList(Collections.singletonList(MII_BASIC_APP_NAME))
+            .appName(MII_BASIC_APP_NAME)),
+        String.format("Failed to create app archive for %s", MII_BASIC_APP_NAME));
+  }
+
+  /**
+   * Create a domain resource with auxiliary image.
+   * Use an domain.yaml file with API Version expliciltly set to v8.
+   * Use the v8 style auxililiary configuration supported in WKO v3.3.x
+   * Start the Operator with latest version
+   * Here the webhook infra started in Operator namespace should implicitly 
+   *  upgrade the domain resource to native k8s format with initContainer 
+   *  configuration in ServerPod section and start the domain 
+   */
+  @Test
+  @DisplayName("Test to implicit upgrade of v8 version of AuxDomain with webhook")
+  void testImplicitAuxV8DomainUpgrade() {
+
+
+    // admin/managed server name here should match with model yaml
+    final String auxiliaryImagePath = "/auxiliary";
+    List<String> archiveList = Collections.singletonList(ARCHIVE_DIR + "/" + MII_BASIC_APP_NAME + ".zip");
+
+    List<String> modelList = new ArrayList<>();
+    modelList.add(MODEL_DIR + "/" + MII_BASIC_WDT_MODEL_FILE);
+    modelList.add(MODEL_DIR + "/model.jms2.yaml");
+
+    // create auxiliary image using imagetool command if does not exists
+    logger.info("creating auxiliary image {0}:{1} using imagetool.sh ", miiAuxiliaryImage, MII_BASIC_IMAGE_TAG);
+    createPushAuxiliaryImageWithDomainConfig(miiAuxiliaryImage, archiveList, modelList);
+
```

Comment:
add a comment here about generating the domain.yaml

Reply:
done